### ISSUE TITLE
th:remove all-but-first + th:each precedence

### ISSUE THREAD

**Issue by dtrunk90** [thymeleaf/thymeleaf#580]
It seems `th:each` will be processed first. If the loop ends up in no elements (e.g. iterable is empty) `th:remove="all-but-first"` will not remove following elements.

Example:
```
<table>
	<tbody th:remove="all-but-first">
		<tr th:each="obj : ${list}"><td>Hello</td></tr>
		<tr><td>World</td></tr>
	</tbody>
</table>
```

If the list is empty this will result in:
```
<table>
	<tbody>
		<tr><td>World</td></tr>
	</tbody>
</table>
```

But I'm expecting:
```
<table>
	<tbody>
	</tbody>
</table>
```

This wasn't the case in Thymeleaf 2.

Version infos:
Thymeleaf 3.0.3.RELEASE
Spring 4.3.7.RELEASE

**Comment by danielfernandez** [comment]
This seems to be the same as #567 

Please could you check that it is fixed for you in `3.0.4-SNAPSHOT`?

**Comment by dtrunk90** [comment]
Yes, it's fixed. Sorry for the duplicate.
